src/Hamiltonians/Molecular.jl

```diff
@@ -0,0 +1,790 @@
+"""
+    MolecularHamiltonian(
+        fcidump_path::String;
+        starting_address::Union{Nothing,FermiFS2C}=nothing,
+        specifier::String="",
+    ) <: AbstractHamiltonian
+    MolecularHamiltonian(
+        fd::ElemCo.FciDumps.FDump;
+        starting_address::Union{Nothing,FermiFS2C}=nothing,
+        specifier::String="",
+    ) <: AbstractHamiltonian
+
+Implements an electronic ab-initio Hamiltonian based on electron overlap integrals in
+FCIDUMP format. It can be used to describe the electronic structure of a molecule
+with fixed nuclear positions (i.e. under the Born-Oppenheimer approximation).
+
+The expression for the Hamiltonian is
+```math
+\\begin{aligned}
+    Ĥ  & = h₀ + Ĥ₁ + Ĥ₂, \\\\
+    Ĥ₁ & = \\sum_{i,j,σ} h_{ij}  a^†_{j,σ} a_{i,σ}, \\\\
+    Ĥ₂ & = ½ \\sum_{i,j,σ,τ} V_{ijkl} a^†_{i,σ} a^†_{j,τ} a_{l,τ} a_{k,σ}.
+\\end{aligned}
+```
+where the constants ``h₀``, ``h_{ij}``, and ``V_{ijkl}`` are provided by the FCIDUMP `fd`.
+See the documentation of [ElemCo.jl](https://elem.co.il) for generating FCIDUMP information
+with a Hartree-Fock calculation, or parsing them from a file.
+
+# Arguments
+* `fcidump_path`: Path to the FCIDUMP file of overlap integrals. This can be generated with
+    standard quantum chemistry packages
+* `fd`: Struct containing FCIDUMP information defined in [ElemCo.jl](https://elem.co.il).
+
+# Keyword Arguments
+* `starting_address::FermiFS2C=nothing`: The starting address (configuration) defines 
+    number of alpha and beta electrons and orbitals.
+* `specifier::String=""`: Arbitrary identifier. It may store the path to the FCIDUMP file or a user
+    specified name.
+
+See also [`FermiFS2C`](@ref)
+
+!!! note
+    The FCIDUMP file is assumed to be a Restricted Hartree-Fock(RHF) Hamiltonian.
+
+!!! note
+    Requires the [ElemCo.jl](https://elem.co.il) package to be loaded with `using ElemCo`.
+"""
+struct MolecularHamiltonian{T,A<:FermiFS2C,D} <: AbstractHamiltonian{T}
+    specifier::String
+    starting_address::A
+    fcidump::D
+end
+
+function MolecularHamiltonian(
+    fcidump_path::String;
+    starting_address::Union{Nothing,FermiFS2C}=nothing,
+    specifier::String="",
+)
+    ext = Base.get_extension(@__MODULE__, :ElemCoExt)
+    if isnothing(ext)
+        error("MolecularHamiltonian requires that ElemCo is loaded, i.e. `using ElemCo`")
+        return nothing
+    end
+    fd = ext.read_fcidump(fcidump_path, Val(4))
+    if specifier == ""
+        specifier = fcidump_path
+    end
+    return MolecularHamiltonian(fd; starting_address=starting_address, specifier=specifier)
+end
+
+function Base.show(io::IO, h::MolecularHamiltonian)
+    io = IOContext(io, :compact => true)
+    print(io, "MolecularHamiltonian(")
+    if ispath(h.specifier)
+        print(io, "\"", h.specifier, "\"")
+    else
+        print(io, typeof(h.fcidump))
```

Comment:
```suggestion
        print(io, "::", typeof(h.fcidump))
```
If you try to type it into the repl, you'll get an error message either way. But indicating that this is a typename may clarify why that is.